src/components/UpcomingQuizCard/UpcomingQuizCard.hook.ts

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from 'react';
+
+import { formatDate } from './UpcomingQuizCard.utils';
+
+export const useUpcomingQuizCard = (date: string | Date) => {
+  const formattedDate = useMemo(() => {
+    const normalizedDate = normalizeDate(date);
+    const nextDay = new Date(normalizedDate);
```

Comment:
The hook constructs a Date from the normalized date then unconditionally adds one day (nextDay.setDate(nextDay.getDate() + 1)) before formatting. This produces an incorrect date (off-by-one). Adjusting dates by +1 to compensate for timezone/parsing issues is a fragile approach and likely a bug.

**Sugerencia:**
```typescript
Do not add +1 day. Use a robust parsing approach (parseISO from date-fns) and then format the parsed date directly. Example fix:

const normalizedDate = normalizeDate(date);
return formatDate(normalizedDate);

Better: normalizeDate should return a valid Date (using parseISO for ISO strings) and formatDate should accept that Date. If timezone normalization is required, handle it explicitly (e.g., startOfDay or use UTC helpers) and document why.
```